src/NuGet.Clients/PackageManagement.VisualStudio/Utility/VsHierarchyUtility.cs

```diff
@@ -37,6 +37,28 @@ public static IVsHierarchy ToVsHierarchy(Project project)
             return hierarchy;
         }
 
+        public static VsHierarchyItem GetHierarchyItemForProject(Project project)
+        {
+            Debug.Assert(ThreadHelper.CheckAccess());
+
+            return new VsHierarchyItem(ToVsHierarchy(project));
+        }
+
+        public static string GetProjectId(Project project)
```

Comment:
>GetProjectId [](start = 29, length = 12)

We need to be careful about changing this. It's possible downstream is not resilient to consuming non-GUID values. Please investigate.